hazelcast/src/main/java/com/hazelcast/internal/probing/ProbeDataSender.java

```diff
@@ -0,0 +1,103 @@
+package com.hazelcast.internal.probing;
+
+import static com.hazelcast.instance.OutOfMemoryErrorDispatcher.inspectOutOfMemoryError;
+import static com.hazelcast.nio.IOUtil.closeResource;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import com.hazelcast.internal.metrics.ProbeLevel;
+import com.hazelcast.internal.probing.ProbeRegistry.ProbeRenderContext;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+import com.hazelcast.util.Clock;
+
+public final class ProbeDataSender implements Runnable, Closeable {
```

Comment:
Should not be part of public API imho. Implementation detail for MC.

Reply:
No. I am not ready with the sending part. I just put it there in the early stages when I wanted to test it.